### DISCUSSION TITLE
Adaptive mesh and InversePowerMethod compatibility

### DISCUSSION THREAD

**Discussion by smpark7** [idaholab/moose#16708] [Q&A General] [answered]
Hi,

Can we run a InversePowerMethod solve with adaptive meshing? The documentation implies that adaptive meshing is only compatible with Transient and Steady solves. I also tried to apply adaptive meshing to my problem both under Executioner and under a separate Adaptivity action but it did not work. MOOSE didn't indicate that the parameters in Adaptivity were unused but no adaptive meshing occurred and the final result is exactly the same as my previous results.

Thanks,
SunMyung

**Comment by loganharbour** [comment, marked as answer]
SunMyung,

The `InversePowerMethod` executioner for eigenvalue problems is soon to be deprecated.

Please take a look at the `Eigenvalue` executioner. You should have no issue with adaptivity with:

```
[Executioner]
  type = Eigenvalue
  solve_type = NONLINEAR_POWER
[]
```

Unfortunately, we're still missing some documentation for this new executioner (but @fdkong should have it up in the future!). Take a look at the tests that use this executioner at the bottom of the page: https://mooseframework.inl.gov/source/executioners/Eigenvalue.html#!

**Reply by fdkong** [reply]
From my understanding, mesh adaptivity does not work with new eigenvalue solvers either. The reason is that nobody asked for this feature  before. You are the first one who asked for this. But it would be straight forward to add this support.

You are welcome to come up with a PR, and I am happy to review and merge the PR. If you do not have resources to do so, you could go ahead to create a ticket. I will add this capability  once I can. Unfortunately, I can not guarantee when I am able to handle the ticket, but I will try the best I can

**Reply by smpark7** [reply]
Thank you both for your replies. I'll have a look at the ```Eigenvalue``` executioner and see if I can tackle this.
